firebase_admin/_messaging_encoder.py

```diff
@@ -0,0 +1,701 @@
+    def check_number_list(cls, label, value):
+        """Checks if the given value is a list comprised only of numbers."""
+        if value is None or value == []:
+            return None
+        if not isinstance(value, list):
+            raise ValueError('{0} must be a list of numbers.'.format(label))
+        non_number = [k for k in value if not isinstance(k, numbers.Number)]
+        if non_number:
+            raise ValueError('{0} must not contain non-number values.'.format(label))
+        return value
+
+    @classmethod
+    def check_analytics_label(cls, label, value):
+        """Checks if the given value is a valid analytics label."""
+        value = _Validators.check_string(label, value)
+        if value is not None and not re.match(r'^[a-zA-Z0-9-_.~%]{1,50}$', value):
+            raise ValueError('Malformed {}.'.format(label))
+        return value
+
+    @classmethod
+    def check_datetime(cls, label, value):
+        """Checks if the given value is a datetime."""
+        if value is None:
+            return None
+        if not isinstance(value, datetime.datetime):
+            raise ValueError('{0} must be a datetime.'.format(label))
+        return value
+
+
+class MessageEncoder(json.JSONEncoder):
+    """A custom JSONEncoder implementation for serializing Message instances into JSON."""
+
+    @classmethod
+    def remove_null_values(cls, dict_value):
+        return {k: v for k, v in dict_value.items() if v not in [None, [], {}]}
+
+    @classmethod
+    def encode_android(cls, android):
+        """Encodes an AndroidConfig instance into JSON."""
+        if android is None:
+            return None
+        if not isinstance(android, _messaging_utils.AndroidConfig):
+            raise ValueError('Message.android must be an instance of AndroidConfig class.')
+        result = {
+            'collapse_key': _Validators.check_string(
+                'AndroidConfig.collapse_key', android.collapse_key),
+            'data': _Validators.check_string_dict(
+                'AndroidConfig.data', android.data),
+            'notification': cls.encode_android_notification(android.notification),
+            'priority': _Validators.check_string(
+                'AndroidConfig.priority', android.priority, non_empty=True),
+            'restricted_package_name': _Validators.check_string(
+                'AndroidConfig.restricted_package_name', android.restricted_package_name),
+            'ttl': cls.encode_ttl(android.ttl),
+            'fcm_options': cls.encode_android_fcm_options(android.fcm_options),
+        }
+        result = cls.remove_null_values(result)
+        priority = result.get('priority')
+        if priority and priority not in ('high', 'normal'):
+            raise ValueError('AndroidConfig.priority must be "high" or "normal".')
+        return result
+
+    @classmethod
+    def encode_android_fcm_options(cls, fcm_options):
+        """Encodes an AndroidFCMOptions instance into a json."""
```

Comment:
Should this be either "a JSON object" or maybe just "JSON?"  Unless `json` is actually a literal.